### DISCUSSION TITLE
Switch from Discord to Matrix

### DISCUSSION THREAD

**Discussion by Wuzzy2** [OpenRCT2/OpenRCT2#20105] [Feature requests] [answered]
I kindly request to replace all Discord servers for development discussion with Matrix rooms. Go to <https://www.matrix.org> to learn more.

## Reason

Discord is proprietary software and is tightly controlled by the company. As a free software developer myself, I do not want to install Discord to be allowed to discuss as I want to minimize the control proprietary software has over me and others. It is deeply concerning that many free software communities have become complicit in allowing this software in their development process. It is a sad irony that in order to get 1st class contact to the devs, you're forced to install proprietary software.

I believe free software development should use free software tools all the way. This includes discussion and contact.

Matrix seems to be a reasonable replacement. 

## Possible objections

* "But everyone uses Discord!": Maybe, but it's still proprietary software, which is bad in itself. The high user count actually makes it worse, as it increases their power. We have to start somewhere.
* "This other FOSS project uses Discord, too!": That just proves my point that FOSS communities have become complicit. This is not a justification to stay complicit. Change is possible. But someone has to start.
* "That's just your personal taste!": No. If Discord would be free software, I would not complain about it. The best solution would be for Discord itself to become FOSS, but we all know this will never happen. Discord is non-free software, and that's the problem. It's also not about me, but about everyone. Injecting Discord into FOSS development teaches other developers and users to become complicit, which is a sign of weakness for the FOSS movement. Another problem is the network effect. The more users are on Discord, the worse their control will become, thus it is reasonable to move users *away* from Discord
* "We can't just shutdown all Discord servers! This will create disruption!": Valid concern. A possible solution is to add a transition period of like 3 months in which both Discord and Matrix runs to allow users to move
* "OK, I'm fine with Matrix, but shutting down Discord servers is way too radical!": And? Linux once used a proprietary software for version control, but then Linus decided to write Git. Since then, nobody sheds a single tear after the old proprietary version control. This was a radical change, too, but for the better.
* "You just want to shill Matrix!": No. I'm not affiliated with the Matrix people in any way. This is my own opinion. I chose Matrix because it is the only viable replacement right now
* "Come on, Discord can't be that bad, right?" https://www.stallman.org/discord.html

**Comment by ZehMatt** [comment]
Should we also move away from GitHub considering the big bad evil Microsoft is behind it?

**Comment by ocalhoun6** [comment]
Is there a good open source alternative to GitHub?

**Comment by Broxzier** [comment]
This post is inviting a debate, which I do not want to go into. We moved from Gitter to Discord for convenience. It has nice integration, is easy to manage, and most importantly, it's a chat program that most gamers already have, making it easy to interact with the community. I'm pretty sure I can speak for everyone on the team that we do not want to move to some other service.

**Comment by Wuzzy2** [comment]
Off-topic: Yes, moving away from GitHub would be nice, too. Unironically. Not because MS, but because it contains a significant amount of proprietary JavaScript code (The JavaScript Trap). (Note: The GitHub site mostly works without JS, but is very buggy.) A possible replacement for GitHub is Codeberg.org (based on Forgejo, a soft fork of Gitea). But yeah, this is off-topic.

Back to topic: Kinda sad this is being shut down instantly with no real discussion. What is wrong with going full FOSS?

**Reply by ZehMatt** [reply]
The issue has been converted to a discussion and its not being shut down, that would mean we close the issue and leave it at that. @Broxzier pointed out already some of the reasons to why we are not going to do that also I personally think that some of the issues you pointed out are not actual issues, I definitely didn't have to pay for Discord and I have my doubt that they sell user information, I'm quite confident the EU would be really mad about that, there is also an fully open source Discord client if you don't trust the official one. Discord is a free service for most of the part, they have subscriptions and other things which they make money with to keep it free for the rest, after all the features you get with Discord are not comparable to matrix, someone has to pay the server bills at the end of the day, nothing is free and when its typically free don't expect anything.

**Reply by Broxzier** [reply]
I'd like to add that the one item on the page Wuzzy linked to about tracking and demanding the user to provide a phone number is not true. I don't see an excessive amount of entries in my pi-hole logs while using Discord, and I've used many different VPN servers while I do not have a phone linked to my account in any way without any problems.

**Comment by 733737** [comment]
we can't go full FOSS because OpenRCT2 requires a licensed game to be installed that you have to pay for.

**Comment by frutiemax** [comment]
I don't think you would convince a lot of people to switch from discord to matrix too easily... Whole RCT communities are built upon discord; it actually brought a lot of visibility for the OpenRCT2 project. Github might also be the most well known host for open-source projects. This also brings new developers that wouldn't have know this project if it would have been hosted on an obscure/less known repository website.

As long as the software used is freeware and the licensing is correct I'm fine using proprietary tools if they save me time.

**Comment by spacek531** [comment]
I don't trust anyone who uses "fed" services. Stay out of my circle glowie.

**Comment by IntelOrca** [comment, marked as answer]
> It is a sad irony that in order to get 1st class contact to the devs, you're forced to install proprietary software.
> I believe free software development should use free software tools all the way. This includes discussion and contact.

You are not forced to do anything. You have contacted us already through GitHub, and you can also contact us via e-mail. Are you unable to use Discord's web client sandboxed with a burner e-mail and discord account for free?

> Maybe, but it's still proprietary software, which is bad in itself.

Why is proprietary software bad? Should all software be free? If so, I would not have my current job which means I could not earn a living. If I could not earn a living, I would not be able to work on OpenRCT2 in my spare time. How would RollerCoaster Tycoon exist if Chris Sawyer was unable to make a living from selling that?

> Discord is proprietary software and is tightly controlled by the company.

How are you measuring control, and what control are you worried about here? Most discussion on OpenRCT2 is open and transparent, and Discord do not currently control, minipulate or censor that discussion. The moment they do, we are free to move to a different platform. The same is true for GitHub.

> "Come on, Discord can't be that bad, right?" https://www.stallman.org/discord.html

Unfortunately these opinions come from an idology I do not completely share.

---

When we moved from Gitter to Discord, our community grew by a large amount and there was more discussion than ever before. This is because the of the large player base that now uses Discord to communicate. In my opinion it is more active than reddit (also proprietary). If we moved to Matrix, we would reduce the discussion considerably, as I doubt you would convince the masses to learn and use Matrix. It would not be convenient for them as they will still need to use Discord for other communities, and DMs (including myself).

On top of that, I think Discord is very good software. It is the best chat/voice client I have ever used. I haven't used Matrix, but I do wonder whether the quality would be as good. Who pays for, or maintains the servers for Matrix? How does Matrix prevent spam, bots, and fake accounts? Are there mobile apps, web clients, and push notifications that work to the same quality? Can Matrix store all past history, images, videos and other media content? Can that past history be easily searched? Does it provide free reliable voice channels, and screen sharing capabilities?

I am definitely not against decentralisation, and I generally encourage it. But right now, I don't think the project currently suffers from having Discord as its official discussion channel. Moving away would cause the project to suffer. Until this is reversed, I see no reason to switch.

**Reply by ocalhoun6** [reply]
>Should all software be free? If so, I would not have my current job which means I could not earn a living.

This is not necessarily so. There are quite a few developers out there who *earn a  living* developing free and open source software.

For example, a lot of the development that goes into Firefox and various Linux distros is actually *paid* work, financed by companies that have a vested interest in improving that software. And there's always the crowdfunding route.

**Reply by IntelOrca** [reply]
@ocalhoun6 I just meant, my current job relies on the software being proprietary. I don't see a functioning world where 100% software could be FOSS. Many libraries can be FOSS and developed by paid developers because they are used by proprietary software.

I'd also like to add that Discord is currently not owned by big tech. so is in my view still an independant company. I think that has some merit.

**Comment by Higgs1** [comment]
Hi everyone,

What's the official Matrix/XMPP/federated chat for OpenRCT2?

Thank you!

**Reply by karst** [reply]
Hi! The only chat platform we are active on is Discord. 
You are always welcome to join us there!
https://discord.gg/openrct2-264137540670324737
